bench/workgen/workgen.cxx

```diff
@@ -2191,6 +2292,22 @@ int WorkloadRunner::run_all(WT_CONNECTION *conn) {
         }
     }
 
+    // Start Idle table cycle thread
+    if (options->max_idle_table_cycle > 0) {
+
+        createDropTableCycle = new WorkloadRunnerConnection();
+        createDropTableCycle->runner = this;
+        createDropTableCycle->connection = conn;
+
+        if ((ret = pthread_create(&idle_table_thandle, NULL, thread_idle_table_cycle_worload,
+            createDropTableCycle)) != 0) {
+            std::cerr << "pthread_create failed err=" << ret << std::endl;
+            std::cerr << "Stopping Create Drop table idle cycle threads." << std::endl;
+            (void)pthread_join(idle_table_thandle, &status);
+            delete createDropTableCycle;
+        }
```

Comment:
There are a couple problems here.  First, if `pthread_create` fails, then `idle_table_thandle` will not be valid, not sure if `pthread_join` would simply fail or crash.   Also `createDropTableCycle` will be deleted, here but since there's no return `createDropTableCycle` will be deleted again further down the function.  Perhaps more importantly, I'd argue that a `pthread_create` error at the beginning of a run feels pretty severe, I'd recommend stopping the run.  No matter what, we need to make sure everything is cleaned up when the function exits.  These same issues exist for the timestamp increment thread above.  Let's get together and talk about some approaches to solve this.

Reply:
Thanks @ddanderson for flagging this. Fixed the pthread_create issue.